src/tests/unit/init_register/CMakeLists.txt

```diff
@@ -4,14 +4,26 @@ target_link_libraries(initialize_mpi OpenCoarrays)
 add_executable(register register.f90)
 target_link_libraries(register OpenCoarrays)
 
-add_executable(register_rename_me register_rename_me.f90)
-target_link_libraries(register_rename_me OpenCoarrays)
+add_executable(register_vector register_vector.f90)
+target_link_libraries(register_vector OpenCoarrays)
 
-add_executable(register_rename_me_too register_rename_me_too.f90)
-target_link_libraries(register_rename_me_too OpenCoarrays)
+add_executable(register_alloc_vector register_alloc_vector.f90)
+target_link_libraries(register_alloc_vector OpenCoarrays)
+
+add_executable(register_alloc_comp_1 register_alloc_comp_1.f90)
+target_link_libraries(register_alloc_comp_1 OpenCoarrays)
+
+add_executable(register_alloc_comp_2 register_alloc_comp_2.f90)
+target_link_libraries(register_alloc_comp_2 OpenCoarrays)
+
+add_executable(register_alloc_comp_3 register_alloc_comp_3.f90)
+target_link_libraries(register_alloc_comp_3 OpenCoarrays)
```

Comment:
Do these tests require GCC trunk/7.x to compile? To run? If so I'll wrap them appropriately for conditional inclusion so long as GCC >= 7